br/pkg/restore/client.go

```diff
@@ -2535,6 +2536,68 @@ func (rc *Client) UpdateSchemaVersion(ctx context.Context) error {
 	return nil
 }
 
+const (
+	alterTableDropIndexFormat      = "ALTER TABLE `%s`.`%s` DROP INDEX `%s`"
+	alterTableAddIndexFormat       = "ALTER TABLE `%s`.`%s` ADD INDEX `%s`(%s)"
+	alterTableAddUniqueIndexFormat = "ALTER TABLE `%s`.`%s` ADD UNIQUE KEY `%s`(%s)"
+	alterTableAddPrimaryFormat     = "ALTER TABLE `%s`.`%s` ADD PRIMARY KEY (%s) NONCLUSTERED"
+)
+
+// RepairIngestIndex drops the indexes from IngestRecorder and re-add them.
+func (rc *Client) RepairIngestIndex(ctx context.Context, ingestRecorder *ingestrec.IngestRecorder, g glue.Glue, storage kv.Storage) error {
+	dom, err := g.GetDomain(storage)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	info := dom.InfoSchema()
+	allSchema := info.AllSchemas()
+	ingestRecorder.UpdateIndexInfo(allSchema)
+	err = ingestRecorder.Iterate(func(_, _ int64, info *ingestrec.IngestIndexInfo) error {
+		var (
+			dropSQL      string = fmt.Sprintf(alterTableDropIndexFormat, info.SchemaName, info.TableName, info.IndexInfo.Name.O)
+			addSQL       string
+			indexTypeSQL string
+			commentSQL   string
+			visibleSQL   string = " VISIBLE"
+		)
+
+		if info.IsPrimary {
+			addSQL = fmt.Sprintf(alterTableAddPrimaryFormat, info.SchemaName, info.TableName, info.ColumnList)
```

Comment:
Should we escape here? Perhaps some guys like to name their indices like(How absurd!):
```SQL
CREATE INDEX `idx``; DROP DATABASE business; --`(some_row);
```

Reply:
done. test updated